### DISCUSSION TITLE
Bug:The Update Page button does not work

### DISCUSSION THREAD

**Discussion by zhongguocn** [WetHat/OnenoteTaggingKit#82] [General]
Windows 11/10 x64
OneNote 365
OneNoteTaggingKit 5.0.8427
===========================
Save the page search results with the "Example" tag to the page, add the "Example" tag on the "A" page, and want to update the saved search results page, but find that the Update Page button does not work.

In addition, the saved search results should be grouped according to the section of the page, which is more convenient to view.

BTW：Hope to add a language option in the settings to allow users to select the interface language of tagkit, rather than the same as that of OneNote.

Thanks.

**Comment by WetHat** [comment]
@zhongguocn  I cannot reproduce your issue (see screenshots below). I've used the `SavedSearchTst` tag for testing. Take note of the timestamps. I've tested both the `Update Page`  button on the ribbon and the `Update` button on the 'Tag Pages' dialog. They both work. 

Which button did you press? You might be running into the issue that updating the `Saved Search` is done be the _Windows Search Service_. When you make a change to a OneNote page it may take a while for the _Windows Search Service_ to re-index the changed OneNote page. If you try to update the `Saved Search`  *now*, does it work?

As for your enhancement requests: 
* _add a language option to settings_ :  Good idea, Can do that.
* _the saved search results should be grouped_ : Also good idea. Can do that.

WetHat

---

|   First Page  tagged  | Second Page tagged
| ---- | -----
| ![First screenshot](https://user-images.githubusercontent.com/10697674/215309111-0777a7e4-15fc-4f81-84ac-ddd985996a3f.png) | ![image](https://user-images.githubusercontent.com/10697674/215309253-0e1871c4-c193-4ed9-bc73-ac96137ee308.png)

**Comment by zhongguocn** [comment]
Thanks  for your reply.

You can't reproduce the bug, probably because I am using Chinese environment and you are English.

This is my Chinese interface
![image](https://user-images.githubusercontent.com/38025627/215324038-bb534441-d9be-4ff0-8143-afa9380146e7.png)


It is suggested to add a link to the saved search page, and users can update the saved search after clicking it.
![image](https://user-images.githubusercontent.com/38025627/215323751-9700dc28-b331-4a03-aa99-f7bfac49fc1e.png)

**Comment by WetHat** [comment]
@zhongguocn,  please try the `Update Page` button on the ribbon one more time and, if the `Saved Search` fails again, send me the TaggingKit logfile in your temporary directory `%temp%`. The filename looks similar to  `taggingkit_8DB020EC8CBDAAB.log`.  Your file will have a different unique id. Maybe I can werkout what is going on.
WetHat

**Comment by zhongguocn** [comment]
thank you. As you said, I tried to click the update page button several times, but it still didn't work.

 taggingkit_8DB02A1A8A1076D.log

> INFO: Register (179) | OneNote Tagging Kit logging activated.
> 	Addin-Version: 5.0.8427.30211
> 	.net Framework Version: 4.0.30319.42000
> 	Configuration: Release
> INFO: OnConnection (102) | Connection mode 'ext_cm_Startup'
> INFO: .ctor (111) | OneNote application proxy constructed successfully
> INFO: OnAddInsUpdate (66) | OneNote Tagging Kit update initiated; Arguments ''
> INFO: Run (56) | Background tagging service started
> INFO: GetCustomUI (174) | UI configuration requested: Microsoft.OneNote
> INFO: OnStartupComplete (152) | Startup Arguments ''
> INFO: OneNoteSchema (259) | Using current notebook to determine OneNote version
> INFO: OneNoteSchema (276) | OneNote schema version is xs2013
> INFO: OnStartupComplete (160) | OneNote Tagging Kit initialization complete!
> INFO: editTags (186) | Show tag editor
> INFO: manageTags (204) | Show settings editor
> INFO: FindPagesByMetadata (402) | Scope = ; metaKey = TaggingKit.PageTags
> INFO: UpdateTagSet (118) | Extracted 256 tags from 1783 pages.
> INFO: editTags (186) | Show tag editor
> INFO: EnqueuePagesForTagging (208) | 1 page(s) enqueued for tagging with 'Savedsearchtst' using SUBTRACT

**Comment by WetHat** [comment]
@zhongguocn,  I think, I found the issue. Please install the attached version of the TaggingKit and let me know `Update Page` now works. If it still does not work, please send the logfile again. 

As for 
_It is suggested to add a link to the saved search page_

As per the discussion in #47, I do not see a feasible way to implement this .

[SetupTaggingKitWiX.zip](https://github.com/WetHat/OnenoteTaggingKit/files/10535542/SetupTaggingKitWiX.zip)

WetHat

**Comment by zhongguocn** [comment]
The update page button does not work after installing the update you provided.
![image](https://user-images.githubusercontent.com/38025627/215478428-5dfd0913-e981-45bb-a20d-60f3ea3c8c08.png)


taggingkit_8DB02FE638C9BAB.log

> INFO: Register (179) | OneNote Tagging Kit logging activated.
> 	Addin-Version: 5.0.8430.22996
> 	.net Framework Version: 4.0.30319.42000
> 	Configuration: Release
> INFO: OnConnection (112) | Connection mode 'ext_cm_Startup'
> INFO: .ctor (111) | OneNote application proxy constructed successfully
> INFO: OnAddInsUpdate (76) | OneNote Tagging Kit update initiated; Arguments ''
> INFO: Run (62) | Background tagging service started
> INFO: GetCustomUI (184) | UI configuration requested: Microsoft.OneNote
> INFO: OnStartupComplete (162) | Startup Arguments ''
> INFO: OneNoteSchema (259) | Using current notebook to determine OneNote version
> INFO: OneNoteSchema (276) | OneNote schema version is xs2013
> INFO: OnStartupComplete (170) | OneNote Tagging Kit initialization complete!
> INFO: manageTags (214) | Show settings editor
> INFO: editTags (196) | Show tag editor
> INFO: Add (48) | Background job scheduled for execution: UNITE Page={68E6E5BE-4DEB-097B-3126-4F5E670B171D}{10}{E19551178505695790650220163957129634464638931}
> INFO: EnqueuePagesForTagging (208) | 1 page(s) enqueued for tagging with '-✩-' using UNITE
> INFO: Execute (108) | Background job executed successfully: UNITE Page={68E6E5BE-4DEB-097B-3126-4F5E670B171D}{10}{E19551178505695790650220163957129634464638931}
> INFO: editTags (196) | Show tag editor
> INFO: Add (48) | Background job scheduled for execution: RESYNC Page={68E6E5BE-4DEB-097B-3126-4F5E670B171D}{10}{E19534053063315610304920121724456568539995061}
> INFO: EnqueuePagesForTagging (208) | 1 page(s) enqueued for tagging with '' using RESYNC
> INFO: FindPagesByMetadata (402) | Scope = ; metaKey = TaggingKit.PageTags
> INFO: UpdateTagSet (118) | Extracted 255 tags from 1780 pages.
> INFO: Execute (108) | Background job executed successfully: RESYNC Page={68E6E5BE-4DEB-097B-3126-4F5E670B171D}{10}{E19534053063315610304920121724456568539995061}
> INFO: Add (48) | Background job scheduled for execution: RESYNC Page={68E6E5BE-4DEB-097B-3126-4F5E670B171D}{10}{E19534053063315610304920121724456568539995061}
> INFO: EnqueuePagesForTagging (208) | 1 page(s) enqueued for tagging with '' using RESYNC
> INFO: FindPagesByMetadata (402) | Scope = ; metaKey = TaggingKit.PageTags
> INFO: UpdateTagSet (118) | Extracted 255 tags from 1780 pages.
> INFO: Execute (108) | Background job executed successfully: RESYNC Page={68E6E5BE-4DEB-097B-3126-4F5E670B171D}{10}{E19534053063315610304920121724456568539995061}
> INFO: editTags (196) | Show tag editor
> INFO: Add (48) | Background job scheduled for execution: UNITE Page={68E6E5BE-4DEB-097B-3126-4F5E670B171D}{10}{E19551178505695790650220163957129634464638931}
> INFO: EnqueuePagesForTagging (208) | 1 page(s) enqueued for tagging with 'Savedsearchtst' using UNITE
> INFO: Execute (108) | Background job executed successfully: UNITE Page={68E6E5BE-4DEB-097B-3126-4F5E670B171D}{10}{E19551178505695790650220163957129634464638931}
>

**Comment by WetHat** [comment]
@zhongguocn , according to the logfile `Update Page` was never executed after you added the 'Savedsearchtst' tag? I do see 2 update requests made from the `Tag Pages dialog`  **before** you added the tag. But no update request after that.

I that what happened? Could you tell me the sequence of events that lead to the problem?

WetHat

**Comment by zhongguocn** [comment]
Operation steps
1. Find the page with the  'Savedsearchtst'  tag and save the search results to the page. 
There are 'First Page' page in the saved search results.

2. Add a 'Savedsearchtst' tag to the 'Second Page' page and click the 'update page' button. 
The search results page has not changed. 'second page 'cannot be seen in search results.

Button 1(update page) cannot update the search results page, but button 2(refresh) can. Can you set the command of button 1 to be the same as that of button 2? I think this can solve the problem.
![image](https://user-images.githubusercontent.com/38025627/215656507-8f08d2d2-b9f4-4188-936c-ba60c3cc59d4.png)

**Comment by WetHat** [comment]
@zhongguocn I think I found the problem. Please check the attached version.

 
[SetupTaggingKitWiX.zip](https://github.com/WetHat/OnenoteTaggingKit/files/10551171/SetupTaggingKitWiX.zip)

**Comment by zhongguocn** [comment]
Thank you for your excellent work. The bug that the 'update page' button is invalid has been solved.
The function of switching plug-in language interface is also available, which is great. Thank you.

BTW：After installing the tagging kit, enter OneNote. When you click 'tag pages' for the first time, you cannot see all the existing page tags. You need to click' setting 'first, and then click' suspended tags'. Then click 'tag pages' to see all existing page tags.
